From ae6dfeb64fd30a4223022323936d28ed0006c155 Mon Sep 17 00:00:00 2001
From: Andrei Botila <andrei.botila@nxp.com>
Date: Fri, 11 Jun 2021 12:53:59 +0300
Subject: [PATCH 06/16] configs: s32: move hardcoded values to the specific
 defconfigs

Hardcoded values in include/configs/s32.h such as CONFIG_BOOTDELAY
and CONFIG_BOOTARGS have been moved in the specific defconfigs for S32.

Issue: ALB-6285

Upstream-Status: Pending

Signed-off-by: Andrei Botila <andrei.botila@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 configs/s32g274a_emu_defconfig      |  1 +
 configs/s32g274a_sim_defconfig      |  1 +
 configs/s32g274abluebox3_defconfig  |  1 +
 configs/s32g274ardb2_defconfig      |  3 ++-
 configs/s32g274ardb2_qspi_defconfig |  3 ++-
 configs/s32g274ardb_defconfig       |  1 +
 configs/s32g274ardb_qspi_defconfig  |  1 +
 configs/s32g2xxaevb_defconfig       |  1 +
 configs/s32g2xxaevb_qspi_defconfig  |  1 +
 configs/s32g398a_emu_defconfig      |  1 +
 configs/s32r45_emu_defconfig        |  1 +
 configs/s32r45_sim_defconfig        |  1 +
 configs/s32r45evb_defconfig         |  1 +
 configs/s32r45evb_qspi_defconfig    |  1 +
 configs/s32v234evb_flash_defconfig  |  1 +
 include/configs/s32.h               | 26 +++++---------------------
 include/configs/s32r45.h            | 16 ----------------
 17 files changed, 22 insertions(+), 39 deletions(-)

diff --git a/configs/s32g274a_emu_defconfig b/configs/s32g274a_emu_defconfig
index 96feb4ca5b..98a957b570 100644
--- a/configs/s32g274a_emu_defconfig
+++ b/configs/s32g274a_emu_defconfig
@@ -12,6 +12,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/configs/s32g274a_sim_defconfig b/configs/s32g274a_sim_defconfig
index 64f72a38ae..d6c035e3b0 100644
--- a/configs/s32g274a_sim_defconfig
+++ b/configs/s32g274a_sim_defconfig
@@ -9,6 +9,7 @@ CONFIG_DISTRO_DEFAULTS=y
 # CONFIG_SYS_MALLOC_F is not set
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/configs/s32g274abluebox3_defconfig b/configs/s32g274abluebox3_defconfig
index b9d2ad4d80..7f981dd8c4 100644
--- a/configs/s32g274abluebox3_defconfig
+++ b/configs/s32g274abluebox3_defconfig
@@ -12,6 +12,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_BOARD_EARLY_INIT_R=y
diff --git a/configs/s32g274ardb2_defconfig b/configs/s32g274ardb2_defconfig
index 8613919f5a..3270ada808 100644
--- a/configs/s32g274ardb2_defconfig
+++ b/configs/s32g274ardb2_defconfig
@@ -11,6 +11,8 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
+CONFIG_BOARD_LATE_INIT=y
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MD5SUM=y
@@ -62,4 +64,3 @@ CONFIG_FSL_DSPI=y
 CONFIG_FSL_QSPI=y
 CONFIG_SJA1105=y
 # CONFIG_EFI_LOADER is not set
-CONFIG_BOARD_LATE_INIT=y
diff --git a/configs/s32g274ardb2_qspi_defconfig b/configs/s32g274ardb2_qspi_defconfig
index d061291faf..7b465f5c51 100644
--- a/configs/s32g274ardb2_qspi_defconfig
+++ b/configs/s32g274ardb2_qspi_defconfig
@@ -12,6 +12,8 @@ CONFIG_DISTRO_DEFAULTS=y
 CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
+CONFIG_BOARD_LATE_INIT=y
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MD5SUM=y
@@ -67,4 +69,3 @@ CONFIG_FSL_DSPI=y
 CONFIG_FSL_QSPI=y
 CONFIG_SJA1105=y
 # CONFIG_EFI_LOADER is not set
-CONFIG_BOARD_LATE_INIT=y
\ No newline at end of file
diff --git a/configs/s32g274ardb_defconfig b/configs/s32g274ardb_defconfig
index b43f82db74..dc0c815515 100644
--- a/configs/s32g274ardb_defconfig
+++ b/configs/s32g274ardb_defconfig
@@ -12,6 +12,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MD5SUM=y
diff --git a/configs/s32g274ardb_qspi_defconfig b/configs/s32g274ardb_qspi_defconfig
index 1c55909541..74031b4dbe 100644
--- a/configs/s32g274ardb_qspi_defconfig
+++ b/configs/s32g274ardb_qspi_defconfig
@@ -13,6 +13,7 @@ CONFIG_DISTRO_DEFAULTS=y
 CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MD5SUM=y
diff --git a/configs/s32g2xxaevb_defconfig b/configs/s32g2xxaevb_defconfig
index 50d5c2595d..7554b55d77 100644
--- a/configs/s32g2xxaevb_defconfig
+++ b/configs/s32g2xxaevb_defconfig
@@ -11,6 +11,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_EEPROM=y
diff --git a/configs/s32g2xxaevb_qspi_defconfig b/configs/s32g2xxaevb_qspi_defconfig
index 623ae59d6a..1fbbb85f59 100644
--- a/configs/s32g2xxaevb_qspi_defconfig
+++ b/configs/s32g2xxaevb_qspi_defconfig
@@ -12,6 +12,7 @@ CONFIG_DISTRO_DEFAULTS=y
 CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_EEPROM=y
diff --git a/configs/s32g398a_emu_defconfig b/configs/s32g398a_emu_defconfig
index e349164348..d8494a2476 100644
--- a/configs/s32g398a_emu_defconfig
+++ b/configs/s32g398a_emu_defconfig
@@ -12,6 +12,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/configs/s32r45_emu_defconfig b/configs/s32r45_emu_defconfig
index 0ac7f5bcf7..a308da3ccf 100644
--- a/configs/s32r45_emu_defconfig
+++ b/configs/s32r45_emu_defconfig
@@ -11,6 +11,7 @@ CONFIG_DISTRO_DEFAULTS=y
 CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon loglevel=7"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/configs/s32r45_sim_defconfig b/configs/s32r45_sim_defconfig
index 7a563c7452..eb8956eeb3 100644
--- a/configs/s32r45_sim_defconfig
+++ b/configs/s32r45_sim_defconfig
@@ -9,6 +9,7 @@ CONFIG_DISTRO_DEFAULTS=y
 # CONFIG_SYS_MALLOC_F is not set
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon loglevel=7"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/configs/s32r45evb_defconfig b/configs/s32r45evb_defconfig
index c5bd83099a..3086e46bbf 100644
--- a/configs/s32r45evb_defconfig
+++ b/configs/s32r45evb_defconfig
@@ -12,6 +12,7 @@ CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_SD_BOOT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon loglevel=7"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_EEPROM=y
diff --git a/configs/s32r45evb_qspi_defconfig b/configs/s32r45evb_qspi_defconfig
index d7d5491a2b..42a9013c2d 100644
--- a/configs/s32r45evb_qspi_defconfig
+++ b/configs/s32r45evb_qspi_defconfig
@@ -13,6 +13,7 @@ CONFIG_DISTRO_DEFAULTS=y
 CONFIG_TOOLS_DEBUG=y
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon loglevel=7"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_EEPROM=y
diff --git a/configs/s32v234evb_flash_defconfig b/configs/s32v234evb_flash_defconfig
index 7f128a60ee..363f4592c2 100644
--- a/configs/s32v234evb_flash_defconfig
+++ b/configs/s32v234evb_flash_defconfig
@@ -7,6 +7,7 @@ CONFIG_NR_DRAM_BANKS=3
 CONFIG_S32_DDR3=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/ram rw earlycon"
 CONFIG_ARCH_EARLY_INIT_R=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_CMD_MEMTEST=y
diff --git a/include/configs/s32.h b/include/configs/s32.h
index 727cd4a1e9..0385f70240 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -251,18 +251,6 @@
 
 #endif
 
-#undef CONFIG_BOOTDELAY
-
-#define CONFIG_BOOTDELAY	3
-
-#define CONFIG_BOOTARGS_LOGLEVEL ""
-
-#undef CONFIG_BOOTARGS
-#define CONFIG_BOOTARGS		\
-	"console=ttyLF" __stringify(CONFIG_FSL_LINFLEX_MODULE) "," __stringify(CONFIG_BAUDRATE) \
-	" root=/dev/ram rw" CONFIG_BOOTARGS_LOGLEVEL " earlycon " \
-	CONFIG_EXTRA_KERNEL_BOOT_ARGS
-
 #define CONFIG_HWCONFIG
 
 #ifdef CONFIG_CMD_BOOTI
@@ -404,8 +392,7 @@
 		"root=/dev/nfs rw " \
 		"ip=${ipaddr}:${serverip}::${netmask}::eth0:off " \
 		"nfsroot=${serverip}:/tftpboot/rfs,nolock,v3,tcp " \
-		"earlycon " CONFIG_EXTRA_KERNEL_BOOT_ARGS \
-		PFENG_EXTRA_BOOT_ARGS "\0" \
+		"earlycon " PFENG_EXTRA_BOOT_ARGS "\0" \
 	"loadtftpimage=tftp ${loadaddr} ${image};\0" \
 	"loadtftpramdisk=tftp ${ramdisk_addr} ${ramdisk};\0" \
 	"loadtftpfdt=tftp ${fdt_addr} ${fdt_file};\0" \
@@ -442,9 +429,8 @@
 				"mmc write ${loadaddr} 0x8 ${fw_sz}; " \
 			"fi; "	\
 		"fi\0" \
-	"mmcargs=setenv bootargs console=${console},${baudrate} " CONFIG_BOOTARGS_LOGLEVEL \
+	"mmcargs=setenv bootargs console=${console},${baudrate}" \
 		" root=${mmcroot} earlycon " \
-		CONFIG_EXTRA_KERNEL_BOOT_ARGS \
 		PFENG_EXTRA_BOOT_ARGS "\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
@@ -468,8 +454,7 @@
 	"netargs=setenv bootargs console=${console},${baudrate} " \
 		"root=/dev/nfs " \
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp " \
-		"earlycon " CONFIG_EXTRA_KERNEL_BOOT_ARGS \
-		PFENG_EXTRA_BOOT_ARGS "\0" \
+		"earlycon " PFENG_EXTRA_BOOT_ARGS "\0" \
 	"netboot=echo Booting from net ...; " \
 		"run netargs; " \
 		"if test ${ip_dyn} = yes; then " \
@@ -491,9 +476,8 @@
 		"else " \
 			"${boot_mtd}; " \
 		"fi;\0" \
-	"flashbootargs=setenv bootargs console=${console}" \
-		CONFIG_BOOTARGS_LOGLEVEL " root=/dev/ram rw earlycon " \
-		CONFIG_EXTRA_KERNEL_BOOT_ARGS \
+	"flashbootargs=setenv bootargs console=${console},${baudrate}" \
+		" root=/dev/ram rw earlycon " \
 		PFENG_EXTRA_BOOT_ARGS ";" \
 		"setexpr uboot_flashaddr " __stringify(UBOOT_FLASH_ADDR) ";" \
 		"setexpr kernel_flashaddr " __stringify(KERNEL_FLASH_ADDR) ";" \
diff --git a/include/configs/s32r45.h b/include/configs/s32r45.h
index ee49e28ee4..8377475467 100644
--- a/include/configs/s32r45.h
+++ b/include/configs/s32r45.h
@@ -18,20 +18,4 @@
 
 #include <configs/s32-gen1.h>
 
-#ifdef CONFIG_TARGET_TYPE_S32GEN1_EMULATOR
-
-#ifdef CONFIG_BOOTARGS_LOGLEVEL
-#undef CONFIG_BOOTARGS_LOGLEVEL
-#endif
-
-#define CONFIG_BOOTARGS_LOGLEVEL	" loglevel=7 "
-
-#ifdef CONFIG_BOOTDELAY
-#undef CONFIG_BOOTDELAY
-#endif
-
-#define CONFIG_BOOTDELAY	-1
-
-#endif /* CONFIG_TARGET_TYPE_S32GEN1_EMULATOR */
-
 #endif
-- 
2.17.1

